src/main/java/org/jenkinsci/plugins/workflow/cps/view/InterpolatedSecretsAction.java

```diff
@@ -117,32 +121,48 @@ public void onLoad(Run<?, ?> run) {
         @Exported
         public String getStepSignature() {
             Map<String, Object> stepArguments;
+            FlowNode node;
             try {
-                stepArguments = getStepArguments(run, nodeId);
+                node = getFlowNode(run, nodeId);
+                ArgumentsAction argumentsAction = node.getPersistentAction(ArgumentsAction.class);
+                if (argumentsAction == null) {
+                    throw new IllegalStateException("null arguments action");
+                }
+                stepArguments = argumentsAction.getArguments();
             } catch (IllegalStateException e) {
                 return "Unable to construct " +  stepName + ": " + e.getMessage();
             }
 
+            if (node instanceof StepNode) {
+                StepDescriptor descriptor = ((StepNode)node).getDescriptor();
+                if (descriptor != null && descriptor.isMetaStep()) {
+                    DescribableParameter p = DescribableModel.of(descriptor.clazz).getFirstRequiredParameter();
+                    if (p != null) {
+                        Object arg = ArgumentsAction.getResolvedArguments(node).get(p.getName());
+                        if (arg instanceof UninstantiatedDescribable) {
+                            return argumentToString(arg);
+                        } else {
+                            return stepName + "(" + argumentToString(arg) + ")";
```

Comment:
I know in the case of `ArchiveArtifacts` the inner argument was an `UninstantiatedDescribable` with a `symbol` so that it would print out the signature correctly. What I was less sure about, was if this was always going to be the case in these scenarios where there is an outer `CoreStep`.